// react-standalone-form/lib/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "FormConsumer", {
  enumerable: true,
  get: function get() {
    return _Form.FormConsumer;
  }
});
Object.defineProperty(exports, "FormThemeProvider", {
  enumerable: true,
  get: function get() {
    return _FormThemeProvider.default;
  }
});
Object.defineProperty(exports, "processField", {
  enumerable: true,
  get: function get() {
    return _helpers.processField;
  }
});
Object.defineProperty(exports, "initiateFormFields", {
  enumerable: true,
  get: function get() {
    return _helpers.initiateFormFields;
  }
});
Object.defineProperty(exports, "updateFieldsRequirements", {
  enumerable: true,
  get: function get() {
    return _helpers.updateFieldsRequirements;
  }
});
Object.defineProperty(exports, "checkboxHandler", {
  enumerable: true,
  get: function get() {
    return _helpers.checkboxHandler;
  }
});
Object.defineProperty(exports, "formIsInvalid", {
  enumerable: true,
  get: function get() {
    return _helpers.formIsInvalid;
  }
});
Object.defineProperty(exports, "getValues", {
  enumerable: true,
  get: function get() {
    return _helpers.getValues;
  }
});
Object.defineProperty(exports, "imageUrltoImageData", {
  enumerable: true,
  get: function get() {
    return _helpers.imageUrltoImageData;
  }
});
Object.defineProperty(exports, "inputHeight", {
  enumerable: true,
  get: function get() {
    return _themeHelpers.inputHeight;
  }
});
Object.defineProperty(exports, "breakpoint", {
  enumerable: true,
  get: function get() {
    return _themeHelpers.breakpoint;
  }
});
Object.defineProperty(exports, "clearfix", {
  enumerable: true,
  get: function get() {
    return _themeHelpers.clearfix;
  }
});
Object.defineProperty(exports, "overlay", {
  enumerable: true,
  get: function get() {
    return _themeHelpers.overlay;
  }
});
Object.defineProperty(exports, "maxRows", {
  enumerable: true,
  get: function get() {
    return _themeHelpers.maxRows;
  }
});
Object.defineProperty(exports, "placeholder", {
  enumerable: true,
  get: function get() {
    return _themeHelpers.placeholder;
  }
});
Object.defineProperty(exports, "withSubmit", {
  enumerable: true,
  get: function get() {
    return _withSubmit.default;
  }
});
Object.defineProperty(exports, "FormControlLogic", {
  enumerable: true,
  get: function get() {
    return _FormControlLogic.default;
  }
});
Object.defineProperty(exports, "withFormControl", {
  enumerable: true,
  get: function get() {
    return _FormControl.default;
  }
});
Object.defineProperty(exports, "defautTheme", {
  enumerable: true,
  get: function get() {
    return _theme.default;
  }
});
Object.defineProperty(exports, "FormActions", {
  enumerable: true,
  get: function get() {
    return _FormActions.default;
  }
});
Object.defineProperty(exports, "FormButton", {
  enumerable: true,
  get: function get() {
    return _FormButton.default;
  }
});
Object.defineProperty(exports, "DefaultButton", {
  enumerable: true,
  get: function get() {
    return _DefaultButton.default;
  }
});
Object.defineProperty(exports, "FormRow", {
  enumerable: true,
  get: function get() {
    return _FormRow.default;
  }
});
Object.defineProperty(exports, "Combine", {
  enumerable: true,
  get: function get() {
    return _Combine.default;
  }
});
Object.defineProperty(exports, "Checkbox", {
  enumerable: true,
  get: function get() {
    return _Checkbox.default;
  }
});
Object.defineProperty(exports, "checkboxTheme", {
  enumerable: true,
  get: function get() {
    return _Checkbox.checkboxTheme;
  }
});
Object.defineProperty(exports, "Checkboxes", {
  enumerable: true,
  get: function get() {
    return _Checkboxes.default;
  }
});
Object.defineProperty(exports, "ImageSelect", {
  enumerable: true,
  get: function get() {
    return _ImageSelect.default;
  }
});
Object.defineProperty(exports, "ImageUpload", {
  enumerable: true,
  get: function get() {
    return _ImageUpload.default;
  }
});
Object.defineProperty(exports, "MultiImageUpload", {
  enumerable: true,
  get: function get() {
    return _MultiImageUpload.default;
  }
});
Object.defineProperty(exports, "Input", {
  enumerable: true,
  get: function get() {
    return _Input.default;
  }
});
Object.defineProperty(exports, "MultiFormInput", {
  enumerable: true,
  get: function get() {
    return _MultiFormInput.default;
  }
});
Object.defineProperty(exports, "MultiSelect", {
  enumerable: true,
  get: function get() {
    return _MultiSelect.default;
  }
});
Object.defineProperty(exports, "Radio", {
  enumerable: true,
  get: function get() {
    return _Radio.default;
  }
});
Object.defineProperty(exports, "Slider", {
  enumerable: true,
  get: function get() {
    return _Slider.default;
  }
});
Object.defineProperty(exports, "Select", {
  enumerable: true,
  get: function get() {
    return _Select.default;
  }
});
Object.defineProperty(exports, "selectTheme", {
  enumerable: true,
  get: function get() {
    return _Select.selectTheme;
  }
});
Object.defineProperty(exports, "Switch", {
  enumerable: true,
  get: function get() {
    return _Switch.default;
  }
});
Object.defineProperty(exports, "switchTheme", {
  enumerable: true,
  get: function get() {
    return _Switch.switchTheme;
  }
});
Object.defineProperty(exports, "Switches", {
  enumerable: true,
  get: function get() {
    return _Switches.default;
  }
});
Object.defineProperty(exports, "TextArea", {
  enumerable: true,
  get: function get() {
    return _TextArea.default;
  }
});
Object.defineProperty(exports, "Loader", {
  enumerable: true,
  get: function get() {
    return _Loader.default;
  }
});
exports.default = void 0;

var _Form = _interopRequireWildcard(require("./core/Form"));

var _FormThemeProvider = _interopRequireDefault(require("./core/FormThemeProvider"));

var _helpers = require("./core/helpers");

var _themeHelpers = require("./core/themeHelpers");

var _withSubmit = _interopRequireDefault(require("./core/withSubmit"));

var _FormControlLogic = _interopRequireDefault(require("./core/FormControlLogic"));

var _FormControl = _interopRequireDefault(require("./default/FormControl"));

var _theme = _interopRequireDefault(require("./core/theme"));

var _FormActions = _interopRequireDefault(require("./default/FormActions"));

var _FormButton = _interopRequireDefault(require("./core/FormButton"));

var _DefaultButton = _interopRequireDefault(require("./default/DefaultButton"));

var _FormRow = _interopRequireDefault(require("./default/FormRow"));

var _Combine = _interopRequireDefault(require("./default/Combine"));

var _Checkbox = _interopRequireWildcard(require("./default/Checkbox"));

var _Checkboxes = _interopRequireDefault(require("./default/Checkboxes"));

var _ImageSelect = _interopRequireDefault(require("./default/ImageSelect"));

var _ImageUpload = _interopRequireDefault(require("./default/ImageUpload"));

var _MultiImageUpload = _interopRequireDefault(require("./default/MultiImageUpload"));

var _Input = _interopRequireDefault(require("./default/Input"));

var _MultiFormInput = _interopRequireDefault(require("./default/MultiFormInput"));

var _MultiSelect = _interopRequireDefault(require("./default/MultiSelect"));

var _Radio = _interopRequireDefault(require("./default/Radio"));

var _Slider = _interopRequireDefault(require("./default/Slider"));

var _Select = _interopRequireWildcard(require("./default/Select"));

var _Switch = _interopRequireWildcard(require("./default/Switch"));

var _Switches = _interopRequireDefault(require("./default/Switches"));

var _TextArea = _interopRequireDefault(require("./default/TextArea"));

var _Loader = _interopRequireDefault(require("./default/Loader"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var _default = _Form.default;
exports.default = _default;
