// react-standalone-form/lib/default/DefaultButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _polished = require("polished");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DefaultButton = function DefaultButton(_ref) {
  var _onClick = _ref.onClick,
      classes = _ref.classes,
      disabled = _ref.disabled,
      children = _ref.children;
  return /*#__PURE__*/_react.default.createElement("button", {
    type: "button",
    className: classes.button,
    onClick: function onClick(e) {
      return _onClick(e);
    },
    disabled: disabled
  }, children);
};

var _default = (0, _reactJss.default)(function (theme) {
  return {
    button: {
      padding: "0 ".concat(theme.sizes.inputHeight / 2, "px"),
      fontSize: theme.typography.inputFontSize,
      lineHeight: "".concat(theme.sizes.inputHeight, "px"),
      height: theme.sizes.inputHeight,
      borderWidth: 0,
      borderRadius: theme.sizes.borderRadius,
      whiteSpace: 'nowrap',
      color: 'white',
      backgroundColor: theme.colors.accent,
      cursor: 'pointer',
      display: 'inline-flex',
      justifyContent: 'center',
      '&:hover': {
        backgroundColor: (0, _polished.lighten)(0.1, theme.colors.accent)
      },
      '&:focus': {
        outline: 'none'
      },
      '&:disabled': {
        opacity: 0.6,
        pointerEvents: 'none'
      }
    }
  };
})(DefaultButton);

exports.default = _default;
