// react-standalone-form/lib/default/FormRow.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _classnames = _interopRequireDefault(require("classnames"));

var _ = require("../.");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var FormRow = function FormRow(_ref) {
  var _classNames;

  var children = _ref.children,
      mobile = _ref.mobile,
      classes = _ref.classes,
      className = _ref.className,
      noBottomGutter = _ref.noBottomGutter;
  return /*#__PURE__*/_react.default.createElement("div", {
    className: (0, _classnames.default)((_classNames = {}, _defineProperty(_classNames, classes.formRow, !mobile), _defineProperty(_classNames, classes.formRowMobile, mobile), _defineProperty(_classNames, classes.noBottomGutter, noBottomGutter), _defineProperty(_classNames, className, className), _classNames))
  }, children);
};

FormRow.propTypes = {
  mobile: _propTypes.default.bool
};

var _default = (0, _reactJss.default)(function (theme) {
  var commonStyling = {
    display: 'flex',
    alignItems: 'flex-end',
    marginBottom: theme.sizes.inputGutterBottom,
    '& .form-control': {
      marginBottom: 0
    },
    '& > div + *': {
      marginLeft: theme.sizes.formRowGutter
    }
  };
  return {
    formRow: _defineProperty({}, (0, _.breakpoint)(theme.breakpoints.sm), commonStyling),
    formRowMobile: commonStyling,
    noBottomGutter: {
      marginBottom: 0
    }
  };
})(FormRow);

exports.default = _default;
