// react-standalone-form/lib/default/Radio.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _classnames = _interopRequireDefault(require("classnames"));

var _ = require("../.");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Radio = function Radio(_ref) {
  var name = _ref.name,
      value = _ref.value,
      required = _ref.required,
      small = _ref.small,
      setValue = _ref.setValue,
      options = _ref.options,
      classes = _ref.classes;
  return /*#__PURE__*/_react.default.createElement("div", null, options.map(function (option, index) {
    var optionLabel = typeof option === 'string' ? option : option.label;
    var optionValue = typeof option === 'string' ? option : option.value;
    return /*#__PURE__*/_react.default.createElement("label", {
      className: (0, _classnames.default)(classes.label, 'form-checkitem', _defineProperty({}, classes.small, small)),
      key: index,
      htmlFor: "".concat(name).concat(index)
    }, /*#__PURE__*/_react.default.createElement("input", {
      type: "radio",
      name: name,
      value: optionValue,
      className: classes.input,
      id: "".concat(name).concat(index),
      checked: optionValue === value,
      onChange: function onChange(e) {
        return setValue(name, e.target.value, required);
      }
    }), " ", optionLabel);
  }));
};

Radio.propTypes = {
  name: _propTypes.default.string.isRequired,
  value: _propTypes.default.string,
  required: _propTypes.default.bool,
  setValue: _propTypes.default.func.isRequired,
  options: _propTypes.default.arrayOf(_propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.shape({
    label: _propTypes.default.string,
    value: _propTypes.default.string
  })]))
};

var _default = (0, _.withFormControl)((0, _reactJss.default)({
  label: {
    display: 'block',
    cursor: 'pointer'
  },
  input: {
    marginRight: 10
  },
  small: {
    fontSize: 12,
    lineHeight: '18px',
    '& $input': {
      marginRight: 5
    }
  }
})(Radio));

exports.default = _default;
