// react-standalone-form/lib/default/TextArea.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactJss = _interopRequireDefault(require("react-jss"));

var _classnames = _interopRequireDefault(require("classnames"));

var _ = require("../.");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var TextArea = function TextArea(_ref) {
  var name = _ref.name,
      value = _ref.value,
      placeholder = _ref.placeholder,
      required = _ref.required,
      setValue = _ref.setValue,
      rows = _ref.rows,
      min = _ref.min,
      classes = _ref.classes;
  return /*#__PURE__*/_react.default.createElement("textarea", {
    className: (0, _classnames.default)(classes.textArea, 'form-input'),
    name: name,
    rows: rows,
    placeholder: placeholder,
    onChange: function onChange(e) {
      return setValue(name, e.target.value, required, {
        min: min
      });
    },
    value: value
  });
};

TextArea.defaultProps = {
  rows: 5
};
TextArea.propTypes = {
  name: _propTypes.default.string.isRequired,
  value: _propTypes.default.string,
  placeholder: _propTypes.default.node,
  required: _propTypes.default.bool,
  setValue: _propTypes.default.func.isRequired,
  rows: _propTypes.default.number,
  min: _propTypes.default.number
};

var _default = (0, _.withFormControl)((0, _reactJss.default)({
  textArea: {
    display: 'block',
    resize: 'vertical'
  }
})(TextArea));

exports.default = _default;
